File: WorkoutTracker.UI/workout-tracker/src/app/models/exercise-in-workout.ts
import { Exercise } from './exercise';

export class ExerciseInWorkout {
  public id: number;
  public exerciseId: number;
  public exerciseName: string;
  public numberOfSets: number;
  public setType: number;
  public exercise: Exercise;
  public resistanceType: number;

  constructor(
    id: number, 
    exerciseId: number, 
    exerciseName: string, 
    numberOfSets: number, 
    setType: number, 
    resistanceType: number) {
      
    this.id = id;
    this.exerciseId = exerciseId;
    //this.exerciseName = exerciseName;
    this.numberOfSets = numberOfSets;
    this.setType = setType;
    this.resistanceType = resistanceType;
    this.exercise = null;
    console.log("ExerciseInWorkout: ", this);
  }
}

File: WorkoutApplication.Domain/Exercises/ExerciseInWorkout.cs
using System;
using System.Collections.Generic;
using System.Text;
using WorkoutApplication.Domain.BaseClasses;

namespace WorkoutApplication.Domain.Exercises
{
    public class ExerciseInWorkout : Entity
    {
        public virtual Exercise Exercise { get; set; }
        public virtual int ExerciseId { get; set; }
        public byte NumberOfSets { get; set; }
        public SetType SetType { get; set; }
        public byte Sequence { get; set; }
    }
}
